be/src/exprs/split_part.cpp

```diff
@@ -170,7 +170,9 @@ StatusOr<ColumnPtr> StringFunctions::split_part(FunctionContext* context, const
             if (split_index(haystack, delimiter, part_number, slice)) {
                 res.append(slice);
             } else {
-                res.append_null();
+                slice.data = haystack.data;
+                slice.size = haystack.size;
+                res.append(slice);
```

Comment:
split_part comes from PostgreSQL, it behaves as follows when not matched:
1. SELECT SPLIT_PART('hello world', '|', 1) returns 'hello world'
2. SELECT SPLIT_PART('hello world', '|', 1) returns empty string ''

do you test case#2? can you add a UT in BE or in sql test to avoid stupid modification of others?

Reply:
Hello @satanson . Thanks for those points. I have tried to adress your suggestions.